### PR TITLE
feat: Modify Modbus simulator to support device scaling

### LINKED ISSUE #161: feat: Modify Modbus simulator to support device scaling
Modify Modbus simulator to support device scaling
- Create a default simulator with port 1502
- Create simulators for scalability test

Requirement and design can refer to https://github.com/edgexfoundry/edgex-taf/issues/137

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #161

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #161 

## What is the new behavior?
- Create a default simulator with port 1502
- Create simulators for scalability test

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: simulator/main.go [hunk 1]
```diff
@@ -9,60 +9,138 @@ package main
...
+	return nil
+}
+
+func scaleDevice(startingPort int, simulatorNumber int) (scaledDevicePorts []int, err error) {
+	log.Printf("Create simulator, startingPort is %d, simulatorNumber is %d \n", startingPort, simulatorNumber)
+	count := 0
+	for count < simulatorNumber {
+		err := createTCPDevice(startingPort)
+		if err != nil {
+			return nil, err
```
Comment (cloudxxx8) [inline simulator/main.go:113]: print out the error message.  you should show which port is unavailable

File: simulator/main.go [hunk 2]
```diff
@@ -9,60 +9,138 @@ package main
...
-	ProtocolTCP = "TCP"
-	ProtocolRTU = "RTU"
-	ModbusPort  = "1502"
+	protocolTCP            = "TCP"
+	protocolRTU            = "RTU"
+	defaultDevicePort      = 1502
+	host                   = "0.0.0.0"
+	startingPortEnvName    = "STARTING_PORT"
+	simulatorNumberEnvName = "SIMULATOR_NUMBER"
+	defaultStartingPort    = 2000
```
Comment (cloudxxx8) [inline simulator/main.go:30]: change the default port to the one you confirm it works

File: simulator/main.go [hunk 3]
```diff
@@ -9,60 +9,138 @@ package main
...
+	config := &serial.Config{
+		Address:  "/tmp/master",
+		BaudRate: 19200,
+		DataBits: 8,
+		StopBits: 1,
+		Parity:   "N",
+	}
+	fmt.Printf("Modbus RTU config %v \n", config)
+	if err := device.ListenRTU(config); err != nil {
+		return err
```
Comment (cloudxxx8) [inline simulator/main.go:100]: print out the error message

File: simulator/main.go [hunk 4]
```diff
@@ -9,60 +9,138 @@ package main
...
-	fmt.Printf("Start up a Modbus %s simulator. \n", *protocol)
 	<-c
+	log.Println("Modbus simulator shutdown.")
+}
+
+func createTCPDevice(port int) error {
+	device := mbserver.NewServer()
+	url := fmt.Sprintf("%s:%d", host, port)
+	if err := device.ListenTCP(url); err != nil {
+		return err
```
Comment (cloudxxx8) [inline simulator/main.go:82]: print out the error message

File: simulator/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+1. Create a default simulator with port 1502
+    ```
+    docker run --rm -p 1502:1502 modbus-simulator
+    ```
+
+2. create simulators for scalability test 
+    ```
+    docker run --rm -d --network host --name modbus-simulator  \
+        -e SIMULATOR_NUMBER=1000 -e STARTING_PORT=10000 modbus-simulator
```
Comment (cloudxxx8) [inline simulator/README.md:21]: modify the example to the real available port number

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: simulator/main.go (hunk 3 above)
Comment (weichou1229) [reply simulator/main.go:100]: I print the error here:
https://github.com/edgexfoundry/device-modbus-go/pull/162/files#diff-a78ce53529228c405845d25a31ffcc9fR60

File: simulator/main.go (hunk 2 above)
Comment (weichou1229) [reply simulator/main.go]: Modified the port as the README file mentioned.

File: simulator/README.md (hunk 5 above)
Comment (weichou1229) [reply simulator/README.md:21]: I tested the port and it work for me.

File: simulator/main.go (hunk 4 above)
Comment (weichou1229) [reply simulator/main.go:82]: I print the error here:
https://github.com/edgexfoundry/device-modbus-go/pull/162/files#diff-a78ce53529228c405845d25a31ffcc9fR60

File: simulator/main.go (hunk 1 above)
Comment (weichou1229) [reply simulator/main.go:112]: The error print at https://github.com/edgexfoundry/device-modbus-go/pull/162/files#diff-a78ce53529228c405845d25a31ffcc9fR142

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**